threejs/fxhash_Rooms/src/index.js

```diff
@@ -101,7 +101,7 @@ const createFloorPlan = () => {
       material
     );
     floorBoundary.rotation.x = figureMap.plane.rot.x;
-    floorBoundary.position.set(figureMap.plane.pos.x, figureMap.plane.pos.y, figureMap.plane.pos.z); // <--- needs to move to a new random position.x/z foreach [i]
+    floorBoundary.position.set(figureMap.plane.pos.x * fxrand(3), figureMap.plane.pos.y, figureMap.plane.pos.z * fxrand(3)); // <--- needs to move to a new random position.x/z foreach [i]
```

Comment:
@Anaglyph-ic is this approach correct here? I'd like each floorBoundary to have a unique x and z position. This way I can leave the figureMap.plane.pos.x / z properties as default + values in the figures.js file. 

Same question when looking at other properties like size. Instead of a single value for x, y size properties, I'd like them to be random. What's the best way to use the figureMap in a forloop when randomness is a requirement? 

![image](https://user-images.githubusercontent.com/54593392/183259388-08bdca35-fba8-4eea-98d0-0f29c8e1683c.png)

Reply:
4 points here here. 

## 1 - `fxrand()` is to be treated as `Math.random()`

`fxrand()` yields a number between 0 and 1. So if you want to have a number between 0 and 3, you need to call `fxrand() * 3`, if you want an integer use `Math.floor(fxrand() * 3)`. `fxrand(3)` will not yield what you think it yields.

## 2 - Randomization happens directly in `figures.js` 

The `fxrand()` function needs to be called in the `figures.js` script. 

So this is how it plays out:

**figure.js**
```js
//plane
  figures.push({
    geometry: {
      type: "PlaneGeometry", // Type of geometry
//
// Randomization of geometry arguments and hence, size
//
      args: [
        // Arguments relevant to the geometry (check THREE API)
        (2 + 5 * fxrand()) * u,
        (2 + 5 * fxrand()) * u,
      ],
    },
//
// This is where the position randomization happens
//
    pos: {
      // Position
      x: fxrand() * 3 * u,
      y: -0.5 * u,
      z: fxrand() * 3  * u,
    },


    rot: {
      // Rotation
      x: -Math.PI * 0.5,
      y: 0,
      z: 0,
    },
    name: "plane",
    lines: true, // Display color segments (like wireframe, but faces not triangles)
    hatch: true, // Fill with white texture
    full: false, // Fill with color texture (in the anaverse, red and cyan)
  });
```

**index.js**
```js
floorBoundary.position.set(figureMap.plane.pos.x, figureMap.plane.pos.y, figureMap.plane.pos.z); 
```

## 3 - What you're looking to achieve would be best achieved by drawing a shape and extruding it.

What you need is a floor plan generator that yields an array of 2D `[[x,y],[x,y],[x,y]]` coordinates. Once this is done, you can use extrude it to make an actual floor mesh, and deduce the position of the walls by interpolating the middle of each side.

## 4 - Best practices

The anaverse uses many "perf hacks" that you probably do not need to use for this project. But if you were perf-oriented, you would iterate on the figures array and call `THREE[figures[i].geometry.type](...figures[i].geometry.args)` and texture on a need-to-update basis. The higher perf hack is to map all geometries by type and arguments so that they can be reused for different meshes. Other techniques can be applied, like workerizing computationally intensive operations.